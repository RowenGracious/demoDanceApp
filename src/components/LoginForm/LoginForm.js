import React from 'react';
import { useForm, FormProvider } from 'react-hook-form';
import { useRouter } from 'next/router';

import InputText from '../common/InputText/InputText';
import PrimaryButton from '../common/Button/PrimaryButton';

import { userData } from '../../utlis/utils';

const LoginForm = () => {
  const methods = useForm();
  const router = useRouter()

  const onSubmit = (data) => {
    //comparing the Form login information with hardcoded user information
    if(userData.emailId === data.emailID.toLowerCase() && userData.password === data.password) {
      //IF true set the localstorage token
      localStorage.setItem('accessToken', Math.random().toString(36));
      return router.push('/dashboard');
    } else {
      alert("Login details incorrect");
    }
  }

  return (
    <FormProvider {...methods}>
      <form className="mx-auto self-center md:w-4/6 w-full p-20 grid grid-cols-1 gap-8 rounded overflow-hidden border-4 border-light-blue-500 border-opacity-25 shadow-lg" onSubmit={methods.handleSubmit(onSubmit)}>
      <h1 className="mx-auto uppercase text-purple-500 font-bold"> Haligonia Dance Academy </h1>
        <h2> Enter Login Information: </h2>
        <InputText 
          type="text" 
          label="Enter your Email ID" 
          name="emailID"
          required="required"
          pattern={{
            value: /\S+@\S+.\S+/,
            message: "Entered value does not match email format"
          }}
        />
        <InputText 
          type="password" 
          label="Enter your Password" 
          name="password"
          required="required"
          minLength= {{
            value: 5,
            message: "min length is 5"
          }}
        />
        <PrimaryButton type="submit"> Continue </PrimaryButton>
      </form>
    </FormProvider>
  );
}

export default LoginForm