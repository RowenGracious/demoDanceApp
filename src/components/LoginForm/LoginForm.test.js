import React from 'react';
import renderer from 'react-test-renderer';
import LoginForm from '../LoginForm/LoginForm';

// snapshot testing for login
test('Login form component', () => {
  const component = renderer.create(
    <LoginForm />
  );
  
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

// Puppeteer testing for login

const puppeteer = require('puppeteer');

let browser
let page

beforeAll(async () => {
  // launch browser 
  browser = await puppeteer.launch(
    {
      headless: true, // headless mode set to false so browser opens up with visual feedback
      slowMo: 0, // how slow actions should be
      defaultViewport : { width : 1200, height : 768 }
    }
  )
  // creates a new page in the opened browser   
  page = await browser.newPage()
})

describe('Successfull Login', () => {
  test('with uppercase email address', async () => {
    await page.goto('http://localhost:3000/')

    const [button] = await page.$x("//a[contains(., 'Login')]");
    if (button) {
        await button.click();
    }
    
    await page.waitForSelector('.mx-auto #emailID')
    await page.type('#emailID', 'JOEDOE@YOPMAIL.COM');
    
    await page.waitForSelector('.mx-auto #password')
    await page.type('#password', 'JoeDoe@12345');
    
    const [signIn] = await page.$x("//button[contains(., 'Continue')]");
    if (signIn) {
        await signIn.click();
    }

    await page.waitForSelector('.makeStyles-root-2 > .MuiButtonBase-root:nth-child(3) > .MuiIconButton-label > .MuiSvgIcon-root > path')
    
    const [logout] = await page.$x("//a[contains(., 'Logout')]");
    if (logout) {
        await logout.click();
    }

    await page.$x("//a[contains(., 'Home')]");

  });
  
  test('with lowercase email address', async () => {
    await page.goto('http://localhost:3000/')

    const [button] = await page.$x("//a[contains(., 'Login')]");
    if (button) {
        await button.click();
    }
    
    await page.waitForSelector('.mx-auto #emailID')
    await page.type('#emailID', 'joedoe@yopmail.com');
    
    await page.waitForSelector('.mx-auto #password')
    await page.type('#password', 'JoeDoe@12345');
    
    const [signIn] = await page.$x("//button[contains(., 'Continue')]");
    if (signIn) {
        await signIn.click();
    }

    await page.waitForSelector('.makeStyles-root-2 > .MuiButtonBase-root:nth-child(3) > .MuiIconButton-label > .MuiSvgIcon-root > path')
    
    const [logout] = await page.$x("//a[contains(., 'Logout')]");
    if (logout) {
        await logout.click();
    }

    await page.$x("//a[contains(., 'Home')]");
  });
});

describe('Unsuccessfull Login', () => {
  test('with incorrect email address', async () => {
    // event listener to catch alertbox and dismiss the alert
    page.on('dialog', async dialog => {
      console.log(dialog.message());
      await dialog.dismiss();
    });

    await page.goto('http://localhost:3000/')

    const [button] = await page.$x("//a[contains(., 'Login')]");
    if (button) {
        await button.click();
    }
    
    await page.waitForSelector('.mx-auto #emailID')
    await page.type('#emailID', 'joedoe123@yopmail.com');
    
    await page.waitForSelector('.mx-auto #password')
    await page.type('#password', 'JoeDoe123@12345');
    
    const [signIn] = await page.$x("//button[contains(., 'Continue')]");
    if (signIn) {
        await signIn.click();
    }

  });
});
// This function occurs after the result of each tests, it closes the browser
afterAll(() => {
  browser.close()
})