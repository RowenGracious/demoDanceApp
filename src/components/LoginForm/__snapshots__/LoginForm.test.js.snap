// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login form component 1`] = `
<form
  className="mx-auto self-center md:w-4/6 w-full p-20 grid grid-cols-1 gap-8 rounded overflow-hidden border-4 border-light-blue-500 border-opacity-25 shadow-lg"
  onSubmit={[Function]}
>
  <h1
    className="mx-auto uppercase text-purple-500 font-bold"
  >
     Haligonia Dance Academy 
  </h1>
  <h2>
     Enter Login Information: 
  </h2>
  <div
    className="grid grid-cols-1 gap-3"
  >
    <label
      htmlFor="emailID"
    >
       
      Enter your Email ID
       
    </label>
    <input
      className="border-2 h-10 p-2"
      id="emailID"
      name="emailID"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
    />
  </div>
  <div
    className="grid grid-cols-1 gap-3"
  >
    <label
      htmlFor="password"
    >
       
      Enter your Password
       
    </label>
    <input
      className="border-2 h-10 p-2"
      id="password"
      name="password"
      onBlur={[Function]}
      onChange={[Function]}
      type="password"
    />
  </div>
  <button
    className="bg-transparent hover:bg-blue-500 text-blue-700 font-semibold hover:text-white py-2 px-4 border border-blue-500 hover:border-transparent rounded"
    type="submit"
  >
     
     Continue 
     
  </button>
</form>
`;
